### ISSUE TITLE
Product price block no more available in patterns

### ISSUE THREAD

**Issue by wpchannel** [woothemes/woocommerce#51138]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The product price block is no more available in patterns. The bug appears since WooCommerce 9.2.3.

### Expected behavior

We should be able to use product price block anywhere.

### Actual behavior

The product price block could not be use anymore in patterns under Full Site Editor. But we can still view it by editing patterns from the hidden post type view (accessible when using a classic theme).
It's only working when editing single product template. Seems related to this: https://github.com/woocommerce/woocommerce/pull/49906

### Steps to reproduce

1. Go to Site Editor
2. Add or edit a pattern
3. Try to add Product Price block

### WordPress Environment

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.2.3
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.1
Log Directory Writable: ✔
WP Version: 6.6.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.58 (Unix) mod_wsgi/4.9.4 Python/3.11 mod_fastcgi/mod_fastcgi-SNAP-0910052141 OpenSSL/1.1.1u
PHP Version: 8.3.9
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.76.1
OpenSSL/1.1.1u

SUHOSIN Installed: –
MySQL Version: 8.0.35
Max Upload Size: 8 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 1
page: 7
post: 2
wp_block: 1
wp_global_styles: 1
wp_navigation: 1
wp_template_part: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 9.2.3

### Inactive Plugins (2) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.3
Hello Dolly: by Matt Mullenweg – 1.7.2

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: 
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 13 B

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/ -  Contains the woocommerce/cart block
Checkout: #9 - /checkout/ -  Contains the woocommerce/checkout block
My account: #10 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.2
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /Users/aurelien/Sites/woocommerce/wp-content/plugins/woocommerce/templates/block-notices/error.php
/Users/aurelien/Sites/woocommerce/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/Users/aurelien/Sites/woocommerce/wp-content/plugins/woocommerce/templates/block-notices/success.php


### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system

Daily Cron: ✔ Next scheduled: 2024-09-05 11:03:48 +00:00
Options: ✔
Notes: 55
Onboarding: skipped

### Action Scheduler ###

Complete: 2
Oldest: 2024-09-04 11:05:22 +0000
Newest: 2024-09-04 11:05:22 +0000

Pending: 1
Oldest: 2024-09-05 11:05:22 +0000
Newest: 2024-09-05 11:05:22 +0000


### Status report information ###

Generated at: 2024-09-04 11:05:32 +00:00
```
</details>

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by kmanijak** [comment]
Hi @wpchannel , 

> Expected behavior: We should be able to use product price block anywhere.

Can I confirm that you're creating a pattern that will later be used as a product layout, e.g. in Products (Beta) or Product Collection block?

Because Product Price and most of the product blocks have ancestors specified for a reason. That makes them available in places that provide the required context for the price to display.

Also, I spotted your comment on [another issue](https://github.com/woocommerce/woocommerce/pull/49906#issuecomment-2328494330):

> Hi there, it seems that we can't add this block to patterns anymore. That's a real issue because we need it in query loop. It breaks on live website.

Please explain your use case with Query Loop block so we can better assess the issue.

**Comment by wpchannel** [comment]
I use product price block in query block to show a grid of product. Then I create a synced pattern like a product card to use it in others query blocks. This way I can edit my product card once and get it updated in all product grids.
This behavior is working fine before this recent update. I can't imagine being not able to do that anymore. For now the quick solution is to use Meta Field plugin to show this information but it's not perfect.

**Comment by kmanijak** [comment]
I strongly recommend using Product Collection block instead of Query Loop. Product Collection is designed specifically to display product grid with lots of capabilities!

Also, I tried the following steps in WooCommerce 9.1.4:
1. Add Product Collection block
2. Create a synced pattern out of Product Template internals
3. Editing the pattern in isolation. I cannot insert Product Price or any other product elements because the required ancestor is not there.

Are those steps the same as your use case (with the difference that I used Product Collection and you use Query Loop)? If not, can you explain step by step how you used that?
Also, what was the previous WooCommerce version in which you could add Product Price anywhere? 🙌

**Comment by wpchannel** [comment]
Sure I can, but a year ago I'm pretty sure that Product Collection block was in beta. Also, I see no information in the user interface or in documentation that advised about it.

I have checked on another WooCommerce website using 9.1.4. It's correct that I can't add Product Price anymore but the block is still showing without any error. With WooCommerce 9.2.3, the block generate an error saying it doesn't exist.

Maybe this new behavior also explains why all query blocks are now in error.

However, being able to create a pattern inside a query either using Query block or Product Collection block is allowed and a best pratice to prevent reinventing the wheel for each grid of posts. So for now, this is a real issue because I can add a Production Collection block with Product Price and then create a pattern.

The problem is not related to the type of query block I'm using but the ability to use some blocks in a pattern.

**Comment by wpchannel** [comment]
Any news about this concern? Right now, we can't add price block in a synchronized pattern. When you have lot of queries in different templates like archives it's a big waste of time.

**Comment by kmanijak** [comment]
> Any news about this concern? Right now, we can't add price block in a synchronized pattern. When you have lot of queries in different templates like archives it's a big waste of time.

Hi @wpchannel, the fix is coming and most likely will be available in WooCommerce 9.5: https://github.com/woocommerce/woocommerce/pull/53034

**Comment by kmanijak** [comment]
Fixed in
- WooCommerce 9.5 - https://github.com/woocommerce/woocommerce/pull/53077
- `trunk` - https://github.com/woocommerce/woocommerce/pull/53108
